scripts/linux/publishReleasePackages.sh

```diff
@@ -166,3 +182,138 @@ if [[ $uploaded == false ]]; then
 else
     echo "Packages Uploaded"
 fi
+
+}
+
+publish_to_github()
+{   
+    #Investigate if this can be derived from a commit, Hardcode for now.
+    if [[ -z $BRANCH_NAME ]]; then
+        echo "No Branch Name Provided"
+        exit 1
+    fi
+    
+    branch_name=${BRANCH_NAME/"refs/heads/"/""}
+    echo "Branch Name is $branch_name"
+     # Get the latest release from a given branch
+    echo "Fetch the latest release: "
+    url="https://api.github.com/repos/Azure/iotedge/releases"
+    header_content="Accept:application/vnd.github.v3+json"
+    header_auth="Authorization:token $GITHUB_PAT"
+    content=$(curl -X GET -H "$header_content" -H "$header_auth" "$url")
+    latest_release=$(echo $content | jq --arg branch "$branch_name" '[.[] | select(.target_commitish==$branch)][0]' | jq '.name')
+    echo "Latest Release is $latest_release"
```

Comment:
Now I remember a bug that I ran into when I used the script. 
We are currently publish both release `1.1.*` and `1.2.*` in the repository. 
Let's say we are doing release 1.2.7, but the latest release is 1.1.8. (ruh roh!!!)
This means the script will think that we are releasing 1.1.9; then proceed to find the changelog info for 1.1.9 which doesn't exist. 

I think the most resilience fix for this is to use parse the branch name `release/1.2` to get the `Major.Minor` then search for the latest version that way.

Reply:
Sorry, but not quite sure I follow 

If we are doing a 1.2.* Release, then the branch Name would be release/1.2. We would get the CHANGELOG from a 1.2 branch and not 1.1. We will also get the latest release from a 1.2 Branch and not a 1.1. Branch. There is no possibility of us getting the latest release tag as a 1.1* release tag if we specify the branch as a 1.2. 

Am I missing something?

Reply:
Nah you've got it. It was me tilting.